src/libraries/Delegation.sol

```diff
@@ -4,14 +4,12 @@ pragma solidity ^0.8.0;
 import {DataType} from '../libraries/DataType.sol';
 
 library Delegation {
-    /// @notice Set the expiry and nonce of the delegatee set by user.
+    /// @notice Set the expiry and nonce of the delegatee.
     function updateAll(DataType.PackedDelegation storage delegated, uint128 expiry, uint128 nonce) internal {
-        uint128 storedNonce;
+        uint256 word;
         unchecked {
-            storedNonce = nonce + 1;
+            word = _pack(expiry, nonce + 1);
```

Comment:
Just notice that the casting of `uint256(nonce)` in `function _pack(...)` is unnecessary.

Reply:
I think the casting is necessary since `nonce` is 128 bits long only. If we don't do the uint256 casting, we will get 0 after shifting 128 bits left.

Reply:
Oh, you're right. Thank you for the explanation.